correctness/correctness_test.go

```diff
@@ -11,155 +11,616 @@ CONDITIONS OF ANY KIND, either express or implied. See the License for the speci
-		DockerImage:       "../resources/timestream-prometheus-connector-docker-image-" + timestream.Version + ".tar.gz",
-		ImageName:         connectorDockerImageName,
-		ConnectorCommands: connectorCMDs,
+	query := "non_existent_metric"
+	resp, err := m.RemoteRead(ctx, query)
+	if err != nil {
+		t.Fatalf("RemoteRead error: %v", err)
+	}
+	if !isEmpty(resp) {
+		t.Errorf("expected empty results for DNE metric but got non-empty")
 	}
+}
 
-	containerIDs = append(containerIDs, integration.StartConnector(t, dockerClient, ctx, connectorConfig))
+func TestWriteNoData(t *testing.T) {
+	ctx := context.Background()
 
-	prometheusConfig := integration.PrometheusContainerConfig{
-		DockerImage: prometheusDockerImage,
-		ImageName:   prometheusDockerImageName,
-		ConfigPath:  prometheusConfigPath,
+	var data []TimeSeriesData // empty
+	err := m.RemoteWrite(ctx, data)
+	if err != nil {
+		t.Fatalf("RemoteWrite returned an error: %v", err)
 	}
-	containerIDs = append(containerIDs, integration.StartPrometheus(t, dockerClient, ctx, prometheusConfig))
+}
 
-	output, err := sendRequest(t, promQL)
-	errorCheck(t, err, dockerClient, ctx)
-	err = writeToFile(output)
-	errorCheck(t, err, dockerClient, ctx)
+func TestWriteNoLabels(t *testing.T) {
+	ctx := context.Background()
 
-	integration.StopContainer(t, dockerClient, ctx, containerIDs)
+	data := []TimeSeriesData{
+		{
+			Labels: map[string]string{},
+			Samples: []SampleData{
+				{Value: 220, Timestamp: time.Now().UnixMilli()},
+			},
+		},
+	}
+	err := m.RemoteWrite(ctx, data)
+	if err == nil {
+		t.Fatal("expected an error (status code 400), but got nil")
+	}
+	if !strings.Contains(err.Error(), "status code 400") {
+		t.Errorf("expected status code 400 error, got: %v", err)
+	}
 }
 
-// loadPromQLFromFile loads the PromQL for correctness testing from the specified filepath.
-func loadPromQLFromFile(t *testing.T, filePath string, queries []string) []string {
-	file, err := os.Open(filePath)
-	require.NoError(t, err)
+func TestWriteNoSamples(t *testing.T) {
+	ctx := context.Background()
 
-	defer file.Close()
+	name := "prometheus_http_requests_total"
+	instance := "mockmetheus"
 
-	scanner := bufio.NewScanner(file)
-	for scanner.Scan() {
-		text := scanner.Text()
-		if len(text) != 0 {
-			queries = append(queries, scanner.Text())
-		}
+	data := []TimeSeriesData{
+		{
+			Labels: map[string]string{
+				"__name__": name,
+				"instance": instance,
+			},
+			Samples: []SampleData{}, // no samples
+		},
 	}
+	if err := m.RemoteWrite(ctx, data); err != nil {
+		t.Fatalf("RemoteWrite error: %v", err)
+	}
+
+	// Wait to ensure ingestion
+	time.Sleep(ingestionWaitTime)
 
-	assert.Nil(t, scanner.Err())
-	return queries
+	// query for within the last 2 seconds, accounting for ingestion delay
+	waitSeconds := int(ingestionWaitTime.Seconds()) + 2
+	query := fmt.Sprintf(`%s{instance="%s"}[%ds]`, name, instance, waitSeconds)
+	resp, err := m.RemoteRead(ctx, query)
+	if err != nil {
+		t.Fatalf("RemoteRead error: %v", err)
```

Comment:
I get the following error when running the tests ONLY with a fresh new table, no matter what `freshTSDB` is set to:

```
correctness_test.go:145: RemoteRead error: status code 400: operation error Timestream Query: Query, https response error StatusCode: 400
```